File: z-clients/python/diom/models/msg_fifo_nack_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgFifoNackIn(BaseModel):
    namespace: str | None = None

    msg_ids: t.List[str]


class _MsgFifoNackIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    msg_ids: t.List[str]

File: z-clients/javascript/src/models/msgFifoNackIn.ts
// this file is @generated

export interface MsgFifoNackIn {
    namespace?: string | null;
    msgIds: string[];
}

export interface MsgFifoNackIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    msgIds: string[];
}

export const MsgFifoNackInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgFifoNackIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            msgIds: object['msg_ids'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgFifoNackIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'msg_ids': self.msgIds,
        };
    }
}